pkg/asset/manifests/aws/zones.go

```diff
@@ -178,45 +200,60 @@ func setSubnetsManagedVPC(in *zonesInput) error {
 		return fmt.Errorf("failed to get availability zones: %w", err)
 	}
 
-	allZones := out.AvailabilityZones()
 	isPublishingExternal := in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy
+	allAvailabilityZones := out.AvailabilityZones()
+	allEdgeZones := out.EdgeZones()
+
 	mainCIDR := capiutils.CIDRFromInstallConfig(in.InstallConfig)
 	in.Cluster.Spec.NetworkSpec.VPC = capa.VPCSpec{
 		CidrBlock: mainCIDR.String(),
 	}
 
 	// Base subnets considering only private zones, leaving one free block to allow
 	// future subnet expansions in Day-2.
-	numSubnets := len(allZones) + 1
+	numSubnets := len(allAvailabilityZones) + 1
 
 	// Public subnets consumes one range from base blocks.
 	if isPublishingExternal {
 		numSubnets++
 	}
 
+	// Edge subnets consumes one CIDR block from private CIDR, slicing it
+	// into smaller depending on the amount edge zones added to install config.
+	if len(allEdgeZones) > 0 {
+		numSubnets++
+	}
+
 	privateCIDRs, err := utilscidr.SplitIntoSubnetsIPv4(mainCIDR.String(), numSubnets)
 	if err != nil {
-		return fmt.Errorf("unable to retrieve CIDR blocks for all private subnets: %w", err)
+		return fmt.Errorf("unable to generate CIDR blocks for all private subnets: %w", err)
+	}
+
+	publicCIDR := privateCIDRs[len(allAvailabilityZones)].String()
```

Comment:
```suggestion
    publicCIDR := ""
    if isPublishingExternal {
	    publicCIDR = privateCIDRs[len(allAvailabilityZones)].String()
	}
```

Reply:
It'll work without the `if` because we have the extra block for future expansion. But I'd rather we use `""` and get an error if we try to use it when public subnets are not used.